### PR TITLE
feat: Add `disableClimateSettingSchedules` prop

### PR DESCRIPTION
**Pull request by xplato**
## Screenshot when disabled

Screenshot 2024-03-06 at 11 19 16 AM

## Video

https://github.com/seamapi/react/assets/87919558/145ffafb-feb6-4611-be39-1d2181a5d912

### CODE REVIEW

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx [hunk 1]
```diff
@@ -45,6 +45,7 @@ export function ThermostatDeviceDetails({
   disableEditAccessCode,
   disableDeleteAccessCode,
   disableResourceIds = false,
+  disableScheduledClimates = false,
```
Comment (razor-x) [inline src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx:48]: These are internal components and should not have the default: the props should all be required. Actually the above default should be removed as well, and fixed in the lock device component too. Having these optional actually resulted in an earlier bug where the prop was not passed down.

```suggestion
  disableScheduledClimates
```

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx [hunk 2]
```diff
@@ -36,12 +37,13 @@ export function DeviceDetails({
     return null
   }
 
-  const props = {
+  const props: Omit<DeviceDetailsProps, 'deviceId'> = {
```
Comment (razor-x) [inline src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:40]: What is this needed now?

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx [hunk 3]
```diff
@@ -36,12 +37,13 @@ export function DeviceDetails({
...
     disableLockUnlock,
     disableDeleteAccessCode,
     disableResourceIds,
     disableCreateAccessCode,
     disableEditAccessCode,
+    disableScheduledClimates,
```
Comment (razor-x) [inline src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:46]:
```suggestion
    disableClimateSettingSchedules,
```

I suspect we will deprecate this  flip the default value to true in the next major version after we introduce `disableScheduledClimateSettings`.

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx (hunk 2 above)
Comment (xplato) [reply src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:40]: It took me a minute to figure out why the prop wasn't being passed through correctly—adding strict typing here will save others this unnecessary delay/investigation. Not required, but not harmful, I think,

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx (hunk 1 above)
Comment (xplato) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Ahh, good point. I've removed this default specifier on all components except for `DeviceDetails` and `DeviceTable`. Are these the two top-level ones that should remain?

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx (hunk 2 above)
Comment (razor-x) [reply src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:40]: Ah ok, I expanded the diff context and I see what's up 👍🏻

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx (hunk 1 above)
Comment (razor-x) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Yes, generally, only defaults on top level components (or UI lib components) should be allowed 👍🏻

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx (hunk 3 above)
Comment (razor-x) [reply src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:46]: There is no Seam API concept called "scheduled climates". Seam only provides "climate setting schedules" which will eventually be deprecated in favor of "scheduled climate settings". Thus, we should name this prop `disableClimateSettingSchedules` to match the Seam API resource. 

When we reach the deprecation stage, we will change the default value of `disableClimateSettingSchedules` to false an introduce `disableScheduledClimateSettings` instead.

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx (hunk 1 above)
Comment (razor-x) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Maybe we should do

```ts
interface ThermostatDeviceDetailsProps extends Required<Pick<CommonProps, '...'>> {
  device: ThermostatDevice
}

Comment (xplato) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: I think this is a better paradigm, yeah. It makes it clearer why global props are being used in context.

File: src/lib/seam/components/DeviceDetails/DeviceDetails.tsx (hunk 3 above)
Comment (xplato) [reply src/lib/seam/components/DeviceDetails/DeviceDetails.tsx:46]: Gotcha—thank you for the explanation!!

File: src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx (hunk 1 above)
Comment (xplato) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: I think this should be done in a separate PR (right after this) for better encapsulation

Comment (xplato) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Ah, I remember why there was a default for `disableResourceIds`: the linter would complain about the nullable boolean. Your type change would fix that, but would be out of the original scope. I can keep it as-is with this default now, or replace the types on this component alone here.

I think I prefer the former, then creating a separate branch to make the type specificity change on all inner-component props. What do you think?

Either is fine—just thinking out loud!

Comment (razor-x) [reply src/lib/seam/components/DeviceDetails/ThermostatDeviceDetails.tsx]: Ok to move it out of scope of this PR.

**Comment by dawnho** [comment]
Could you show a screenshot of what it looks like with this enabled?

**Comment by xplato** [comment]
> Could you show a screenshot of what it looks like with this enabled?

@dawnho uploaded screenshot and video—thank you for reminding me!

**Review by razor-x — APPROVED** [review]

**Pull request merged**